src/ICodegenFactory.hack

```diff
@@ -307,4 +307,22 @@ interface ICodegenFactory {
    * @see codegenType
    */
   public function codegenNewtype(string $name): CodegenType;
+
+  /**
+   * Generate a class of trait xhp attribute.
+   *
+   * @see codegenXHPAttributef
+   */
+  public function codegenXHPAttribute(string $name): CodegenXHPAttribute;
```

Comment:
This breaks all current implementers of this interface out in the wild.
What do we think about this breaking change?

Reply:
Unavoidable: adding new language features to codegen needs to be acceptable